app/src/main/java/com/lucas/medicaltools/MainActivity.kt

```diff
@@ -1,11 +1,87 @@
 package com.lucas.medicaltools
 
+import android.annotation.SuppressLint
+import android.app.SearchManager
+import android.content.Context
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.Menu
+import android.view.View
+import android.widget.ProgressBar
+import androidx.appcompat.widget.SearchView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.databinding.ActivityMainBinding
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.schedulers.Schedulers
+import mu.KotlinLogging
+import java.util.*
 
+private val logger = KotlinLogging.logger {}
 class MainActivity : AppCompatActivity() {
+    private lateinit var medicalTools: List<MedicalTool>
+    private lateinit var medicalAdapter: MedicalToolAdapter
+    private lateinit var searchView: SearchView
+    private lateinit var binding: ActivityMainBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
```

Comment:
Remember to add the delegate implementation to ensure all connects. Ensure to read a mosby guide or check the Android app code for the Mosby delegate in the AppCompatBaseActivity.